applications: serial_lte_modem: Fix MQTT QoS2 (#14091)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/MQTT_AT_commands.rst

::

   #XMQTTEVT=<evt_type>,<result>

* The ``<evt_type>`` value is an integer indicating the type of the event.
  It can return the following values with #XMQTTCON:

  * ``0`` - Acknowledgment of connection request (CONNACK).
  * ``1`` - Disconnection notification (DISCONNECT).
    The MQTT client is disconnected from the MQTT broker once this event is notified.

* The ``<result>`` value is an integer. It can return the following values:

  * ``0`` - Success.
  * *Negative value* - Error code indicating the reason for the failure.

Unsolicited notification
------------------------

::
   #XMQTTEVT=<evt_type>,<result>

* The ``<evt_type>`` value is an integer indicating the type of the event.

>   It can return the following values after the connection is established:

Author: I think that all the events that can be seen after the connection, not directly related to any other operation would go in here.

>   It can return the following values after the connection is established:

Author: I ended up breaking the event listing in the beginning as having them below each command is easier to understand.

> * The ``<evt_type>`` value is an integer indicating the type of the event.
>   It can return the following values after the connection is established:
> 
>   * ``9`` - Ping response from the MQTT broker (PINGRESP).
>     This is received when we ping (PINGREQ) the broker after the keep alive is reached.
> 

Reviewer: ```suggestion
* The ``<evt_type>`` value is an integer indicating the type of the event.
  After the connection is established, it can return ``9`` to indicate a ping response from the MQTT broker (PINGRESP).
  This is received when pinging (PINGREQ) the broker after the keep alive is reached.
```

   #XMQTTEVT: <evt_type>,<result>

* The ``<evt_type>`` value is an integer.
  It can return the following values for the ``#XMQTTUNSUB`` command:

>   * ``8`` - Acknowledgment of the unsubscription request (UNSUBACK).

Reviewer: ```suggestion
  It can return ``8`` for the ``#XMQTTUNSUB`` command to indicate an acknowledgment of the unsubscription request (UNSUBACK).
